osu.Game/Beatmaps/BeatmapSetInfo.cs

```diff
@@ -35,8 +35,15 @@ public class BeatmapSetInfo : IHasPrimaryKey, IHasFiles<BeatmapSetFileInfo>, ISo
         [NotMapped]
         public BeatmapSetMetrics Metrics { get; set; }
 
+        /// <summary>
+        /// The beats per minute of this beatmap set's song.
+        /// </summary>
+        public double BPM { get; set; }
```

Comment:
Why are you storing the aggregate BPM to the set while storing the length to the beatmap? why not just use the same dynamic aggregation and store the BPM in the beatmap along with the length?

Reply:
what do you think about restoring this one back to `OnlineInfo` and create another property inside the beatmap info (`BeatmapInfo.BPMMode`)

Reply:
sure.

Reply:
now that we store `BPM` inside `BeatmapInfo`, should i sort by BPM inside the beatmaps?

Reply:
I doubt it. Sounds less useful than whatever sort it would use otherwise.